### PR TITLE
Friend request handling on UserProfile modal

### PR DESCRIPTION
**Pull request by maartenRamadan**

### CODE REVIEW

File: vivid-server/src/modules/friends/friends.service.ts [hunk 1]
```diff
@@ -28,10 +28,10 @@ export class FriendsService {
 
   // Add FriendEntity to database (=> send friend request)
   async sendFriendRequest(
-    user_1: string,
-    user_2: string,
-    requested_by: string,
-    requested_to: string,
+    user_1: UserEntity,
+    user_2: UserEntity,
```
Comment (mrjvs) [inline vivid-server/src/modules/friends/friends.service.ts]: dont use entities here, they are hard to get and you only need the ids anyway
(also this break the comparison check on line 37)

File: vivid-server/src/modules/friends/friends.service.ts [hunk 2]
```diff
@@ -108,15 +108,18 @@ export class FriendsService {
...
-      .andWhere('user_1 = :u1 OR user_2 = :u2', {
+      .where('user_1 = :u1 AND user_2 = :u2', {
         u1: userId,
-        u2: userId,
+        u2: friendId,
+      })
+      .orWhere('user_1 = :u3 AND user_2 = :u4', {
+        u3: friendId,
+        u4: userId,
       })
```
Comment (mrjvs) [inline vivid-server/src/modules/friends/friends.service.ts]: do a comparison check to see which order they need to be (`user_1 < user_2` is always true)

File: vivid-client/src/components/styled/modals/UserProfile.modal.tsx [hunk 3]
```diff
@@ -78,6 +79,114 @@ function BlockAction(props: { userData: any; userId: string }) {
...
+            userId: props.friendId,
+          },
+        ],
+      });
+    }
+    if (unFriend.done) {
+      unFriend.reset();
+      props.userData?.updateUser({
+        friends: props.userData.user.friends.filter(
+          (v: any) => v !== friendship,
```
Comment (mrjvs) [inline vivid-client/src/components/styled/modals/UserProfile.modal.tsx]: friendship is null here

**Review by mrjvs — CHANGES_REQUESTED** [review]

**Review by mrjvs — APPROVED** [review]

**Review by mrjvs — CHANGES_REQUESTED** [review]
Code looks good, although the approach for the friend request modal needs to change:
 1. request all friendship data when you open the modal (useFetch with onLoad: true)
 2. user input must be a username. not userid (that will require changes in the backend)
 3. dont request userdata on input of the textbox, that will spam the server. do it on button click

**Review by mrjvs — APPROVED** [review]

**Review by mrjvs — APPROVED** [review]

**Review by westerbarendregt — APPROVED** [review]

**Pull request merged**